File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestPercent.rs
println!("getPercent");
let getPercent: f64 = callback("getPercent", &[]);
println!("getPercent: {}", getPercent);
callback_void("setPercent", &[param(getPercent)]);

println!("getPercentVector");
let getPercentVector: Vec<f64> = callback("getPercentVector", &[]);
println!("getPercentVector: {:?}", getPercentVector);
callback_void("setPercentVector", &[param(getPercentVector)]);

println!("getPercentVectorAsList");
let getPercentVectorAsList: Vec<f64> = callback("getPercentVectorAsList", &[]);
println!("getPercentVectorAsList: {:?}", getPercentVectorAsList);
callback_void("setPercentVectorAsList", &[param(getPercentVectorAsList)]);

println!("getPercentMatrix");
let getPercentMatrix: Vec<Vec<f64>> = callback("getPercentMatrix", &[]);
println!("getPercentMatrix: {:?}", getPercentMatrix);
callback_void("setPercentMatrix", &[param(getPercentMatrix)]);

println!("getPercentMatrixAsList");
let getPercentMatrixAsList: Vec<Vec<f64>> = callback("getPercentMatrixAsList", &[]);
println!("getPercentMatrixAsList: {:?}", getPercentMatrixAsList);
callback_void("setPercentMatrixAsList", &[param(getPercentMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestPercent.hs
putStrLn ( show ( "getPercent" ) )
getPercent :: Double <- callback "getPercent" []
putStrLn ( show ( getPercent ) )
callback "setPercent" [param getPercent] :: IO ()

putStrLn ( show ( "getPercentVector" ) )
getPercentVector :: [Double] <- callback "getPercentVector" []
putStrLn ( show ( getPercentVector ) )
callback "setPercentVector" [param getPercentVector] :: IO ()

putStrLn ( show ( "getPercentVectorAsList" ) )
getPercentVectorAsList :: [Double] <- callback "getPercentVectorAsList" []
putStrLn ( show ( getPercentVectorAsList ) )
callback "setPercentVectorAsList" [param getPercentVectorAsList] :: IO ()

putStrLn ( show ( "getPercentMatrix" ) )
getPercentMatrix :: [[Double]] <- callback "getPercentMatrix" []
putStrLn ( show ( getPercentMatrix ) )
callback "setPercentMatrix" [param getPercentMatrix] :: IO ()

putStrLn ( show ( "getPercentMatrixAsList" ) )
getPercentMatrixAsList :: [[Double]] <- callback "getPercentMatrixAsList" []
putStrLn ( show ( getPercentMatrixAsList ) )
callback "setPercentMatrixAsList" [param getPercentMatrixAsList] :: IO ()
